libnopegl/src/node_colorstats.c

```diff
@@ -281,20 +281,14 @@ static int init_block(struct colorstats_priv *s, struct gpu_ctx *gpu_ctx)
     /* Colorstats needs to write into the block so we bind it as SSBO */
     s->blk.usage = NGLI_BUFFER_USAGE_STORAGE_BUFFER_BIT;
 
-    s->blk.buffer = ngli_buffer_create(gpu_ctx);
-    if (!s->blk.buffer)
-        return NGL_ERROR_MEMORY;
-
     /*
      * The size of the buffer depends on the texture size, which will only be
      * known in the update callback for textures fed by a Media node. We use a
      * variadic buffer size because don't want to wait for the texture update to
-     * compile the shaders and prepare the pipelines.
-     *
-     * The init and prepare callbacks are still too early for this buffer
-     * allocation, but we still need to set its usage for the pipeline bindings.
+     * compile the shaders and prepare the pipelines. We also flag this block
+     * as dynamic so its parents know that they need to watch for buffer changes.
      */
-    s->blk.buffer->usage |= NGLI_BUFFER_USAGE_STORAGE_BUFFER_BIT;
+    s->blk.dynamic = 1;
```

Comment:
Why do you need a dynamic flag if you have a "buffer_id" that serve as an update flag? (Not convinced about `buffer_id` name btw, but maybe I misunderstood its goal)

Reply:
Changed to buffer_rev and dropped the dynamic flag.